src/README.md

```diff
@@ -0,0 +1,32 @@
+## Extract Method
+
+To automatically build unvalidated language files, perform the following steps:
+
+* Ensure all validated languages have been included in the `src/validated-languages.json` file.
+  * This JSON file must only include the ISO 639-1 language tag (i.e. the language folder name).
+  * Any IETF language tags that do not match a folder name (e.g. `de-CH` or `fr-CA`) will be ignored.
+
+    ```js
+    [
+      "de",
+      "es",
+      "tr",
+      "ar"
+    ]
+    ```
+
+* For variants:
+  * The extract script will automatically compare the CLDR data between a root language and language variant.
+  * At the time of this writing, no variant files have been determined to be necessary; but `de-CH` (Swiss Standard German) is the only known variant was determined to be necessary and has been manually added.
+  * If you determine that a language variant needs to be included, add its IETF language tag to the `src/extract-variants.json` file to force this script to build an "unvalidated" template file for that variant. For example, adding `fr-CA`, the French Canadian language variant to the template file will automatically build a `ca.json` file within the `fr` folder.
+
+    ```js
+    [
+      "fr-CA"
+    ]
+    ```
```

Comment:
I'm confused here too. On the one hand you're saying it'll automatically determine variants (`will automatically compare the CLDR data between a root language and language variant.`). On the other hand you need to manually specify the IETF tag to generate a variant file? And this variant file is then unvalidated, against the meaning of the validated-languages.json?

Reply:
The extract method compares the variant data to the root language data. It's not perfect, which is why you manually added `de-CH`. So if you determined that you wanted to add a new variant, adding the IETF tag to the `extract-variants.json` file would force the method to create an "unvalidated" template for you - instead of manually copying the root language and adding the two extra `variant` and `variantNative` entries. Of course you don't have to use this method, but I also included it to make sure that the language variant file would be properly built.

Reply:
This information isn't clearly documented in my opinion.

Reply:
One question not regarding the doc; If the task is able to generate a variant if it's included in extract-variants.json, why isn't it possible to do that by default?

Reply:
It can do it by default, but when the root language has the same diacritics as the variant, it skips creating the file... I'm not sure why it missed `de-CH`

Reply:
By default would be the optimal solution, without the need to specify which variants you need. You just need all variants if they differ from the root language.

Reply:
I'm double checking now...

Reply:
Okay please let me know when I can review again.

Reply:
It looks like the problem is because the extract method is only looking for diacritics, and `ß` is not a diacritic. Otherwise it appears to be working as expected.

Reply:
What's it then?

Reply:
Oh, I did find another issue... give me a minute to fix it.

Reply:
Sure, let me know when you're done.

Reply:
Ok, I fixed a problem with the extract method completely missing character clusters... I also had to add one exception for the [`nl` language](https://github.com/unicode-cldr/cldr-misc-full/blob/master/main/nl/characters.json#L12) due to them including a combining diacritic in the language example (`{íj\\u0301}`)... this means our database won't include `j́` but I think it'll work out once (if ever) someone validates the language... or maybe I can attempt it.

Reply:
Don't understand that 😆 However, ready for a review?

Reply:
Yep